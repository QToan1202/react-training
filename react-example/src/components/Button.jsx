import React from 'react'
import logProps from '../HOCs/logProps'

class Button extends React.Component {
  render() {
    return (
      <button>{this.props.title}</button>
    )
  }
}

export default logProps(Button)
