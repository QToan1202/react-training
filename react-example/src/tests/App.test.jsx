import { render, screen } from "@testing-library/react";
import Button from "../components/Button";
import { describe, it, expect } from "vitest";
import '@testing-library/jest-dom'

describe("App", () => {
  it("renders button", async () => {
    render(<Button title="hello" />);
    const button = await screen.findByRole("button");

    expect(button).toMatchSnapshot()
    expect(button).not.toBeNull();
    expect(button).not.toBeDisabled();
    expect(button).toHaveTextContent("hello");
    expect(button).toBeInTheDocument()
  });
});
