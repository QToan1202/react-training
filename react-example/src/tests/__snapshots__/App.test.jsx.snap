// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders button 1`] = `
<button>
  hello
</button>
`;
